src/reconcile/rec_write.c

```diff
@@ -5731,12 +5743,24 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
 			mod->mod_disk_image = r->multi->disk_image;
 			r->multi->disk_image = NULL;
 			mod->mod_page_las = r->multi->page_las;
-		} else
+		} else {
 			WT_RET(__wt_bt_write(session, r->wrapup_checkpoint,
 			    NULL, NULL, NULL,
 			    true, F_ISSET(r, WT_REC_CHECKPOINT),
 			    r->wrapup_checkpoint_compressed));
 
+			/*
+			 * Checkpoint timestamps are saved in the Btree handle,
+			 * copied into the checkpoint when it completes.
+			 */
+			btree->oldest_start_ts = r->multi->addr.oldest_start_ts;
+			btree->newest_start_ts = r->multi->addr.newest_start_ts;
+			btree->newest_stop_ts = r->multi->addr.newest_stop_ts;
```

Comment:
I thought this would be an aggregated value, something like 
```
if (btree->oldest_start_ts > r->multi->addr.oldest_start_ts)
    btree->oldest_start_ts = r->multi->addr.oldest_start_ts;
```

Could you help me understand?